packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -891,15 +901,11 @@ class RenderOpacity extends RenderProxyBox {
 
   @override
   void paint(PaintingContext context, Offset offset) {
-    if (child != null) {
-      if (_alpha == 0) {
-        // No need to keep the layer. We'll create a new one if necessary.
-        layer = null;
-        return;
-      }
-      assert(needsCompositing);
-      layer = context.pushOpacity(offset, _alpha, super.paint, oldLayer: layer as OpacityLayer?);
+    if (_alpha == 0) {
+      layer = null;
```

Comment:
This asymmetry is strange. Paint is no longer in charge of creating the layer, but if it is no longer needed, it is in charge of removing it?

(I sorta expected that updateCompositedLayer would just return null if no layer is needed anymore, not sure how practical that is, though)

Reply:
I think we could handle this automatically. Consider if `_wasRepaintBoundary` is true and `isRepaintBoundary` is false. Then instead of rebuilding from this layer, we rebuild from a parent (ensuring we have an attached layer). Then before we hit _repaintCompositedChild, we null out the layer and do a regular paint?

Reply:
Sounds reasonable. We should document that as the lifecycle of the returned layer on updateCompositiedLayer.

Reply:
I wasn't imaging that we'd call updateCompositiedLayer, just we'd null out the layer ourselves

Reply:
Yes, that sounds good.

Reply:
Done